plugins/woocommerce/client/legacy/css/admin.scss

```diff
@@ -5565,45 +5582,533 @@ img.help_tip {
 	}
 }
 
+/*
+ * Match product & coupon editor page
+ * This selector is increasing specificity a bit too much,
+ * but we need to contain the style update to product and coupon editor only.
+ */
+.post-php.post-type-product,
+.post-new-php.post-type-product,
+.post-php.post-type-shop_coupon,
+.post-new-php.post-type-shop_coupon {
+	// Link color.
+	a,
+	.button-link  {
+		color:  var(--wp-admin-theme-color, $studio-wordpress-blue);
+	}
+
+
+	/* Style all buttons */
+	.button,
+	.wp-media-buttons .button,
+	#screen-meta-links .show-settings {
+		background: none;
+		padding: 0 $gap-smaller;
+		margin: 0;
+		display: inline-flex;
```

Comment:
I'll address it after extracting rules to a separate file, as the order and specificity may be changed there

Reply:
This is the biggest downside of scoping the changes to the product editor page only with `.post-php.post-type-product` - it significantly increases the specificity of all selectors. Introducing the problems like this one.

I can scope out this particular instance, but it's hard to predict and prevent all the other cases.

Reply:
Alternatively, I could not set the `display` property and leave it as it was styling it as `inline-block` (set by wp core).
It will help to prevent drastic failures like this one when the hidden element is shown.

However, the margin-based styling of internals is more flaky IMO, plus it will not prevent the other styles being overwritten for the rules with specificity lower than 0-3-0

Reply:
I removed `display: flex` in https://github.com/woocommerce/woocommerce/pull/52632/commits/4ce8050217b223d0b10de88c192cb82d300706db so it should solve at least the problem of non-hideable elements. 

However, it has a a little downside, as the whitespace in inline items do not fully collapse (to 0 width), which makes the gap betweenthe  media icon and the text one space wider:
![image](https://github.com/user-attachments/assets/2b48658b-f66a-47ab-a5f8-ce82cf276bec)